webapp/channels/src/components/user_settings/display/manage_timezones/manage_timezones.tsx

```diff
@@ -111,7 +113,8 @@ export default class ManageTimezones extends React.PureComponent<Props, State> {
             timezone,
         };
 
-        actions.updateMe(updatedUser).
+        const action = this.props.adminMode ? this.props.actions.patchUser : this.props.actions.updateMe;
```

Comment:
Similar to some of my other suggestions, I wonder if passing along these actions as props would also make sense.

I'm starting to think that having a `UserSettingsContext` that contains `adminMode`, the user, their preferences, and actions to update the user/preferences could've made this a lot easier. Unfortunately, I don't think that would work well with `mapStateToProps` (another win for React Redux's hooks!)